ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/searchtree/algorithms/SearchTree.java

```diff
@@ -224,8 +212,8 @@ private void updateOptimalLeafWithNextDepthBestLeaf(AbstractNetworkPool networkP
             TECHNICAL_LOGS.info("Leaves to evaluate: {}", numberOfCombinations);
         }
         AtomicInteger remainingLeaves = new AtomicInteger(numberOfCombinations);
-        List<ForkJoinTask<Object>> tasks = naCombinationsSorted.keySet().stream().map(naCombination ->
-            networkPool.submit(() -> optimizeOneLeaf(networkPool, naCombination, naCombinationsSorted, remainingLeaves))
+        List<ForkJoinTask<Object>> tasks = naCombinationsSorted.stream().map(naCombination ->
+            networkPool.submit(() -> optimizeOneLeaf(networkPool, naCombination, remainingLeaves))
```

Comment:
Actually, the `naCombinationsSorted` was only used to get the boolean associated to `naCombination` since it is a map. Now, this boolean is computed with an independent method.

However, the call to `optimizeOneLeaf` comes from a stream on `naCombinationsSorted` which is now a TreeSet so the order in which the leaves are processed is still reproducible.